### DISCUSSION TITLE
Where is the reference coordinate system of the /odom data located after undocking the turtlebot4?

### DISCUSSION THREAD

**Discussion by WaitingForAll** [turtlebot/turtlebot4#305] [Q&A]
I an wondering where the reference coordinate system from the odometry data is located, right after the undocking routine of the turtlebot 4. I observed the following by watching the odometry data published:

Green coordinate system should be the observed referance coordinate frame. Black the coordinate system from the tb4:
![image](https://github.com/turtlebot/turtlebot4/assets/139748006/3befe696-64eb-4f43-b369-3555f102abfa)

Is this everytime the reference coordinate system for the odometry data?

I am asking, because i had the case, that the reference coordiante system was oriented like the one after undocking of the tb4 (black, right hand side) and not like the reference coordinate frame in the picture at the dock (green).

**Comment by bmaxdk** [comment]
After undocking, the reference coordinate system of the `/odom` data for the TurtleBot 4 is typically located at the point where the robot was turned on or where it started its current operation. This is the default behavior in most robotic systems using odometry.

Odometry in robotic systems keeps track of the robot's movement relative to its starting point. When the robot starts moving from its dock or initial position, the `/odom` frame's origin is set at this starting location. As the robot moves, the odometry data updates to reflect the robot's new position relative to this origin point.
